api/src/main/java/marquez/db/OpenLineageDao.java

```diff
@@ -86,7 +88,37 @@ void createLineageEvent(
       String producer);
 
   @SqlQuery("SELECT event FROM lineage_events WHERE run_uuid = :runUuid")
-  List<LineageEvent> findOlEventsByRunUuid(UUID runUuid);
+  List<LineageEvent> findLineageEventsByRunUuid(UUID runUuid);
+
+  @SqlQuery(
+      """
+  SELECT event
+  FROM lineage_events le
+  WHERE (le.event_time < :before
+  AND le.event_time >= :after)
+  ORDER BY le.event_time DESC
+  LIMIT :limit""")
+  List<LineageEvent> getAllLineageEvents(ZonedDateTime before, ZonedDateTime after, int limit);
+
+  /**
+   * This is a "hack" to get inputs/outputs namespace from jsonb column: <a
+   * href="https://github.com/jdbi/jdbi/issues/1510#issuecomment-485423083">explanation</a>
+   */
+  @SqlQuery(
+      """
+  SELECT le.event
+  FROM lineage_events le, jsonb_array_elements(coalesce(le.event -> 'inputs', '[]'::jsonb) || coalesce(le.event -> 'outputs', '[]'::jsonb)) AS ds
```

Comment:
Is the  `jsonb` required? It kills the idea of `namespace` index and is kind of corner-case scenario.

Reply:
I think the idea of getting events from some dataset namespace is not a corner case scenario, but very common one. For example, getting all events that relate to particular Postgres database.

Reply:
Also, this would be fast in the future, after @wslulciuc split of facets. Then we could just filter by that AFAIK.

Reply:
Ok. If so, the index `lineage_events(job_namespace, event_time DESC);` will not help and can be removed, isn't it?

Reply:
FYI @julienledem @wslulciuc - this query won't work, since it requires full `lineage_events` table scan on every call, together with this `jsonb` unnesting.

I've talked with @pawel-big-lebowski and we tried to find faster query. The best we've come to is this:

```sql
WITH job_events AS (
    SELECT le.event
    FROM lineage_events le
    WHERE le.job_namespace = :namespace
    AND (le.event_time < :before
    AND le.event_time >= :after)
    ORDER BY le.event_time DESC
), dataset_events AS (
    SELECT le.event, le.event_time
    FROM lineage_events le
    JOIN dataset_versions dv on le.run_uuid = dv.run_uuid
    JOIN datasets_view ds on dv.dataset_uuid = ds.uuid
    JOIN namespaces n on ds.namespace_uuid = n.uuid
    WHERE n.name = :namespace
    AND (le.event_time < :before
    AND le.event_time >= :after)
    ORDER BY le.event_time DESC
)
SELECT le.event
FROM (
    SELECT * FROM dataset_events
    UNION ALL
    SELECT * FROM job_events
) le
ORDER BY le.event_time
LIMIT :limit
```

It's relatively fast - compared to previous version, which needed to do full sequential scan, it will only need to get all the events that have `dataset_versions` joined to namespace - which is a fraction compared to full table scan. It would be possible to make that faster if we could "predicate push down" filters on time. `created_at` is Marquez processing time field - we can't use that.

However, it's relatively pointless. Above query won't take into account events, where one of the input datasets has given namespace, and we don't see a fast way to get them with current database schema. 

We could either

1. get rid of the namespace filtering altogether, leaving only "get all events" endpoint
2. leave very inefficient endpoint, which will need to sequentially scan all the lineage_events table
3. use above query, which will be much faster, but won't include input datasets. 

Thoughts?

Reply:
I decided to remove this feature, and restore it if needed in future PR.